packages/react/src/components/TreeView/TreeView.js

```diff
@@ -0,0 +1,135 @@
+/**
+ * Copyright IBM Corp. 2016, 2018
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React, { useEffect, useRef, useState } from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import { settings } from 'carbon-components';
+import { keys, match, matches } from '../../internal/keyboard';
+
+const { prefix } = settings;
+
+export default function TreeView({
+  children,
+  className,
+  multiselect,
+  onChange,
+  selected: preselected = [],
+  size = 'default',
+  ...rest
+}) {
+  const treeClasses = classNames(className, `${prefix}--tree`, {
+    [`${prefix}--tree--${size}`]: size !== 'default',
+  });
+  const treeRootRef = useRef(null);
+  const treeWalker = useRef(treeRootRef?.current);
+  const [selected, setSelected] = useState(preselected);
+  const handleSelect = (event, { value } = {}) => {
+    if (multiselect && event.metaKey) {
+      if (!selected.includes(value)) {
+        setSelected(selected.concat(value));
+      } else {
+        setSelected(selected.filter(selectedValue => selectedValue !== value));
+      }
+    } else {
+      setSelected([value]);
+    }
+    if (onChange) {
+      onChange(event, { value: value?.value });
```

Comment:
I got a simple string as `value` here, my apologies if I'm missing something.

```suggestion
      onSelect(event, { value });
```

BTW I think we should change the new list of selection (same one as the first arg of `setSelected()`) if we want to use `onChange()` name. My apologies if I missed this in our conversation.